ext/nokogiri/html4_sax_parser_context.c

```diff
@@ -83,7 +83,7 @@ parse_with(VALUE self, VALUE sax_handler)
   }
 
   Data_Get_Struct(self, htmlParserCtxt, ctxt);
-  Data_Get_Struct(sax_handler, htmlSAXHandler, sax);
+  TypedData_Get_Struct(sax_handler, htmlSAXHandler, &noko_sax_handler_type, sax);
```

Comment:
This one was surprising to us, it's cast as a `htmlSAXHandler`, but it's the same type actually. This is why we named the type `noko_sax_handler_type` without specializing either HTML or XML.

Reply:
:+1: